signature-aggregator/aggregator/cache/cache.go

```diff
@@ -0,0 +1,65 @@
+package cache
+
+import (
+	"math"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/pingcap/errors"
+	"go.uber.org/zap"
+)
+
+type Cache struct {
```

Comment:
There is a new goroutine for each [http.Serve](https://pkg.go.dev/net/http#Serve) request handled. We should add a lock to the cache to provide mutual exclusion.

Reply:
The LRU implementation currently in use says this in [its docs](https://pkg.go.dev/github.com/hashicorp/golang-lru/v2): "All caches in this package take locks while operating, and are therefore thread-safe for consumers."